rclcpp/include/rclcpp/node_interfaces/node_base.hpp

```diff
@@ -129,6 +136,7 @@ class NodeBase : public NodeBaseInterface
   std::vector<rclcpp::CallbackGroup::WeakPtr> callback_groups_;
 
   std::atomic_bool associated_with_executor_;
+  std::function<void()> call_executor_function_;
```

Comment:
Perhaps a better name for this would be `notify_executor_callback_` or `notify_executor_of_callback_group_change_function_` or something. We're technically calling the executor here, but that doesn't indicate why we're calling it.

I like `notify_executor_of_callback_group_change_function_` the best of the two above, but something more descriptive like that, even if it is not that exactly is what I'd like to see.

Reply:
Maybe even better, since this class really shouldn't be coupled to a executor (creates a conceptual cycle), just `notify_of_callback_group_change_function_` might be even better. It doesn't matter who we're notifying, we're just notifying them.